backend/src/main/java/com/hackner/musicompass/service/ArtistService.java

```diff
@@ -1,34 +1,58 @@
 package com.hackner.musicompass.service;
+import com.hackner.musicompass.db.ArtistMongoDb;
 import com.hackner.musicompass.discogsapi.model.DiscogsArtist;
 import com.hackner.musicompass.discogsapi.model.DiscogsArtistSearchResults;
 import com.hackner.musicompass.model.Artist;
 import com.hackner.musicompass.discogsapi.service.DiscogsArtistApiService;
 import lombok.Builder;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+
 import java.util.Optional;
 
 @Service
 @Builder
 public class ArtistService {
     private final DiscogsArtistApiService discogsArtistApiService;
+    private final ArtistMongoDb artistMongoDb;
 
     @Autowired
-    public ArtistService(DiscogsArtistApiService discogsArtistApiService) {
+    public ArtistService(DiscogsArtistApiService discogsArtistApiService, ArtistMongoDb artistMongoDb) {
         this.discogsArtistApiService = discogsArtistApiService;
+        this.artistMongoDb = artistMongoDb;
     }
 
-    public Optional<Artist> getArtistByArtistName (String artistName) {
-        DiscogsArtistSearchResults DiscogsArtistSearchResults = discogsArtistApiService.getDiscogsArtistByArtistName(artistName);
+    public Optional<Artist> getArtistBySearchTerm (String searchTerm) {
+        DiscogsArtistSearchResults DiscogsArtistSearchResults = discogsArtistApiService.getDiscogsArtistByArtistName(searchTerm);
         if (DiscogsArtistSearchResults.getResults().length == 0){
             return Optional.empty();
         }
-        DiscogsArtist discogsArtist = DiscogsArtistSearchResults.getResults()[0];
+
+        DiscogsArtist[] discogsArtistResultArray = DiscogsArtistSearchResults.getResults();
```

Comment:
```suggestion
        DiscogsArtist[] discogsArtistResult = discogsArtistSearchResults.getResults();
```

Reply:
this will be changed in the future anyway ....